libs/execution_context/ebpf_link.c

```diff
@@ -90,6 +90,10 @@ ebpf_link_initialize(
         NULL,
         NULL);
 
+    if (return_value != EBPF_SUCCESS) {
+        goto Exit;
+    }
+
     if (!provider_data) {
         return_value = EBPF_INVALID_ARGUMENT;
         goto Exit;
```

Comment:
See https://github.com/microsoft/ebpf-for-windows/pull/539/files#diff-c14a8d075874a74f3dc279088def67e69e031f00016092c8da47dba0bb4513a1 
What you have is ok, but I also removed 97-99 as being useless after the fix.
Feel free to incorporate that here or just leave it for my PR.